### PR TITLE
Update markdown runtime files

### PR DESCRIPTION
**Pull request by tpope**
This is the long overdue update to my markdown runtime files. Thanks!

### CODE REVIEW

File: runtime/ftplugin/markdown.vim [hunk 1]
```diff
@@ -11,16 +11,28 @@ runtime! ftplugin/html.vim ftplugin/html_*.vim ftplugin/html/*.vim
...
+
+if get(g:, 'markdown_recommended_style', 1)
+  setlocal expandtab tabstop=4 softtabstop=4 shiftwidth=4
+endif
+
+if !exists("g:no_plugin_maps") && !exists("g:no_markdown_maps")
+  nnoremap <silent><buffer> [[ :<C-U>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "bsW")<CR>
+  nnoremap <silent><buffer> ]] :<C-U>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "sW")<CR>
+  xnoremap <silent><buffer> [[ :<C-U>exe "normal! gv"<Bar>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "bsW")<CR>
+  xnoremap <silent><buffer> ]] :<C-U>exe "normal! gv"<Bar>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "sW")<CR>
```
Comment (UncleBill) [inline runtime/ftplugin/markdown.vim:30]: I think we can use `<Cmd>` here.
```suggestion
  nnoremap <silent><buffer> [[ <Cmd>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "bsW")<CR>
  nnoremap <silent><buffer> ]] <Cmd>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "sW")<CR>
  xnoremap <silent><buffer> [[ <Cmd>exe "normal! gv"<Bar>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "bsW")<CR>
  xnoremap <silent><buffer> ]] <Cmd>exe "normal! gv"<Bar>call search('\%(^#\{1,5\}\s\+\S\\|^\S.*\n^[=-]\+$\)', "sW")<CR>
```

Comment (tpope) [reply runtime/ftplugin/markdown.vim:30]: I don't think that buys us anything, and I am highly reticent to make a last minute change.

**Comment by brammool** [comment]
I'll include it, thanks.  I'll update the date in the syntax file to today.  I see the one in the filetype plugin is set to tomorrow :-)

**Comment by brammool** [comment]
Running checks I notice that the ftplugin needs to set 'cpoptions' to Vim values, because of the pattern used in the mapping.

**Comment by brammool** [comment]
> Thanks for the date fix, not sure how I flubbed that. Do you want me
> to send a `'cpoptions'` save/restore change or are you already
> handling it?

I already added it.  Just make sure your copy also has it, so that when
you send me an update it won't disappear.

-- 
Q: What do you call a fish without an eye?
A: fsh!
Q: What do you call a deer with no eyes?
A: no eye deer.
Q: What do you call a deer with no eyes and no legs?
A: still no eye deer.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by tpope** [comment]
Will do, thanks!

**Pull request closed without merging**